Marlin/src/HAL/ESP32/i2s.cpp

```diff
@@ -148,29 +157,48 @@ void stepperTask(void *parameter) {
     xQueueReceive(dma.queue, &dma.current, portMAX_DELAY);
     dma.rw_pos = 0;
 
+    const bool using_ftMotion = TERN0(FT_MOTION, ftMotion.cfg.mode);
+
     while (dma.rw_pos < DMA_SAMPLE_COUNT) {
-      if (!nextMainISR) {
-        Stepper::pulse_phase_isr();
-        nextMainISR = Stepper::block_phase_isr();
-      }
-      #if ENABLED(LIN_ADVANCE)
-        else if (!nextAdvanceISR) {
-          Stepper::advance_isr();
-          nextAdvanceISR = Stepper::la_interval;
-        }
-      #endif
-      else
-        i2s_push_sample();
 
-      nextMainISR--;
+      #if ENABLED(FT_MOTION)
 
-      #if ENABLED(LIN_ADVANCE)
-        if (nextAdvanceISR == Stepper::LA_ADV_NEVER)
-          nextAdvanceISR = Stepper::la_interval;
+        if (using_ftMotion) {
+          if (!nextMainISR) {
+            nextMainISR = FTM_MIN_TICKS;
+            stepper.ftMotion_stepper();
+            endstops.update();
+            TERN_(BABYSTEPPING, if (babystep.has_steps()) stepper.babystepping_isr());
+          }
+          nextMainISR = 0;
+        }
```

Comment:
@narno2202 — This is just crudely copied from `stepper.cpp` so I'm sure more needs to be done here. For instance I didn't insert a call to `i2s_push_sample()` or think very hard about how `nextMainISR` is being used here.